src/converse-profile.js

```diff
@@ -257,9 +257,23 @@ converse.plugins.add('converse-profile', {
             },
 
             afterRender () {
+                this.updateStatusDisplayForSelfContact();
                 this.renderAvatar();
             },
 
+            updateStatusDisplayForSelfContact () {
+                const self = this;
+                // changing the status-display of the self-contact after the user changed the xmpp-status
+                const contacts = _converse.roster.models;
+                _.each(contacts, function(contact) {
```

Comment:
You can use `contacts.forEach` instead of `_.each` (I'm trying to move away from lodash as much as possible).

Also, `function(contact)` can be an arrow function: `contact => {`, then you don't need to use `self`.

Reply:
Looking at the code further, instead of looping through all contacts, you can instead do

 ```
const self_contact = contacts.findWhere({'jid': _converse.connection.jid});
if (self_contact) {
    self_contact.presence.save('show', this.model.get('status') || 'offline');
}
```

However, I don't think a View is the right place to put this kind of code.

Look here how I listen for the `add` event for a roster contact and then do stuff in the listener.
https://github.com/conversejs/converse.js/blob/master/src/headless/converse-roster.js#L935

You can do the same kind of thing and then use the code above. Alternatively you can add an `initialize` method to [_converse.RosterContacts](https://github.com/conversejs/converse.js/blob/master/src/headless/converse-roster.js#L391) and then create a listener there for the `add` event.

Either approach is probably fine, but because your listener depends on `converse-profile`, it's probably better to take the first approach and to put the code inside `converse-profile.js`.

Then, whenever the contact is added (after fetching it from cache or from the XMPP server), the right `status` value will be set.

Also, you should probably use `save` instead of `set` so that the value gets persisted in the cache.

Then, another thing to think about is to register an event handler for when `status` changes, so that you can update the contact's presence.